File: 0274-h-index/0274-h-index.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        sorted_citations = sorted(citations, reverse=True)

        if (sorted_citations[0] <= 0):
            return 0

        if (len(citations) == 1):
            return 1
        
        paper_with_citation_count = 0
        for i, citation in enumerate(sorted_citations):
            if (citation == 0):
                return paper_with_citation_count
            else :
                paper_with_citation_count +=1

            if (i + 1 >= citation):
                if (citation > paper_with_citation_count -1):
                    return citation
                else :
                    return paper_with_citation_count -1

        return paper_with_citation_count

File: 0274-h-index/0274-h-index.java
class Solution {
    public int hIndex(int[] citations) {
        int n = citations.length;
        int[] count = new int[n + 1]; 
        
        for (int c : citations) {
            if (c >= n) {
                count[n]++; // Jika lebih dari n, masukkan ke count[n]
            } else {
                count[c]++;
            }
        }
        
        int h = 0;
        for (int i = n; i >= 0; i--) {
            h += count[i]; // Akumulasi jumlah paper
            if (h >= i) {
                return i; // H-index ditemukan
            }
        }
        
        return 0;
    }
}